File: BuddhaGameServer/src/network/socket/actions/PackageListener.java
package network.socket.actions;

import java.util.HashSet;

public abstract class PackageListener {
    private HashSet<ClientPackage> mSubscribePackages = new HashSet<>();

    public void subscribePackages(ClientPackage clientPackage)
    {
        mSubscribePackages.add(clientPackage);
    }

    public void unsubscribePackages(ClientPackage clientPackage)
    {
        mSubscribePackages.add(clientPackage);
    }

    public boolean isSubscribe(ClientPackage clientPackage)
    {
        return mSubscribePackages.contains(clientPackage);
    }

    public abstract void onReceive(ClientPackage clientPackage) throws Exception;
}

File: BuddhaGameClient/Assets/Scripts/Network/Socket/PackageListener.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace BuddhaGame
{
	public abstract class PackageListener 
	{
		private HashSet<ServerPackage> mSubscribePackages = new HashSet<ServerPackage>();

		public void subscribePackages(ServerPackage package)
        {
			mSubscribePackages.Add(package);
		}

		public void unsubscribePackages(ServerPackage package)
		{
			mSubscribePackages.Remove(package);
		}

		public bool isSubscribe(ServerPackage package)
        {
			return mSubscribePackages.Contains(package);			
        }

		public abstract void onReceive(ServerPackage package);
	}
}
